app/util/autoversioning.py

```diff
@@ -1,90 +1,32 @@
+import functools
 import os
 import subprocess
 
-from app.util import fs
-
-
 _MAJOR_MINOR_VERSION = '0.5'
 
-_calculated_version = None  # We will cache the calculated version so that it can't change during execution.
-_VERSION_FILE_PATH = os.path.join(os.path.dirname(__file__), 'package_version.py')
-_VERSION_FILE_BACKUP_PATH = os.path.join(os.path.dirname(__file__), 'package_version.py.bak')
-
 
+@functools.lru_cache(maxsize=1)
 def get_version():
     """
     Get the version of the application. This method should return the correct version in both the
-    packaged and local (running inside the git repo) cases.
+    packaged and local (running inside the git repo) cases. A valid version string must be returned
+    to prevent execution and/or build errors. Version detection will fail in shallow-cloned repo.
 
     :return: The version of the application
     :rtype: str
     """
-
-    return _get_frozen_package_version() or _calculate_source_version() or '0.0.0'
-
-
-def _try_rename(src, dst):
-    try:
-        os.rename(src, dst)
-        return True
-    except (FileExistsError, FileNotFoundError):
-        # Skip backing up the original package_version.py if a FileExistsError or FileNotFoundError happened.
-        # FileExistsError might happen on Windows as NTFS doesn't support writing to a file while the file
-        # is opened in python.
-        return False
-
-
-def _try_remove(src):
-    try:
-        os.remove(src)
-    except OSError:
-        pass
-
-
-def write_package_version_file(package_version_string):
-    """
-    Write the specified version string to package_version.py. This method is intended to be called
-    during the process of freezing a package for release. This in-effect hard codes the version into
-    the dist package.
-
-    This also backs up the original file, which can be restored with another method in this module.
-
-    :param package_version_string: The version to write to the file -- presumably the output of
-                                   get_version()
-    :type package_version_string: str
-    """
-    package_version_file_contents = 'version = "{}"  # DO NOT COMMIT\n'.format(package_version_string)
-
-    _try_rename(_VERSION_FILE_PATH, _VERSION_FILE_BACKUP_PATH)  # Backup the original file.
-    fs.write_file(package_version_file_contents, _VERSION_FILE_PATH)
-
-
-def restore_original_package_version_file():
-    """
-    Restore the backed up version of package_version.py. This is just a convenience method to help
-    us remember not to commit changes to the package version file.
-    """
-    if not _try_rename(_VERSION_FILE_BACKUP_PATH, _VERSION_FILE_PATH):
-        # Ensure the temp file is removed if there was no backup copy.
-        os.remove(_VERSION_FILE_PATH)
+    return _calculate_source_version() or _get_frozen_package_version() or '0.0'
```

Comment:
I feel like this is still overly complicated. The reason right now is so that the developer can see the CR version change in their local repo and have a "-mod" value append. Maybe this is not necessary. Then all you need do is:

call _calculate_source_verison() in setup.py when you are building the pex file
all other code calls _get_frozen_packge_version() which reads the valid version from pex
if #2 fails because you're in a local repo, then the version string could fallback to default value or a timestamp.
The current version generation is in the reverse order from above, so that the developer will always see the latest calculated version in their repo.